consensus/src/threshold_simplex/actors/voter/actor.rs

```diff
@@ -1934,6 +2058,163 @@ impl<
+                        work.is_empty(),
+                        |(min, current, sender, msg)| {
+                            min_view = min;
+                            latest = current;
+                            let view = msg.view();
+                            let verifiable = msg.message(&namespace);
+                            messages.insert(counter, (sender, msg, verifiable.len()));
+                            for (namespace, message, signature) in verifiable {
+                                work.entry(view)
+                                    .or_default()
+                                    .entry((namespace, message))
+                                    .or_default()
+                                    .entry(signature)
+                                    .or_default()
+                                    .push(counter);
+                            }
+                            counter = counter.wrapping_add(1);
+                        },
+                    )
+                    .await
+                    {
+                        return;
+                    }
+
+                    // Attempt to do something in current view
+                    let (view, (namespace, message), signatures) =
+                        if let btree_map::Entry::Occupied(mut entry) = work.entry(latest) {
+                            let records = entry.get_mut();
+                            let ((namespace, message), signatures) = records.pop_first().unwrap();
+                            if records.is_empty() {
+                                entry.remove();
+                            }
+                            (latest, (namespace, message), signatures)
+                        } else {
+                            let newest = *work.keys().next_back().unwrap();
+                            let btree_map::Entry::Occupied(mut entry) = work.entry(newest) else {
+                                unreachable!("missing view");
+                            };
+                            let records = entry.get_mut();
+                            let ((namespace, message), signatures) = records.pop_first().unwrap();
+                            if records.is_empty() {
+                                entry.remove();
+                            }
+                            (newest, (namespace, message), signatures)
+                        };
+
+                    // If nothing to do in current, take next highest (may be ahead of current view)
+                    trace!(
+                        view,
+                        current = latest,
+                        min_view,
+                        count = signatures.len(),
+                        "performing batch verification"
+                    );
+                    batch_size.observe(signatures.len() as f64);
+                    let identity = supervisor.identity(view).unwrap();
+                    let result = partial_verify_multiple_public_keys::<V, _>(
```

Comment:
Unless it is a `notarize` from the leader (needed to start `peer_proposal`), we should wait to perform a batch verification on any message until we've seen `2f+1` (it is useless to us before then?).